### PR TITLE
Draft: Typescript: minor fixes

### PR DESCRIPTION
**Pull request by chippieTV**
- [x] confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!

Just some type cleanup

### CODE REVIEW

**Comment by HarelM** [comment]
Thanks! These look great! Can you help with the other comments about bad formatting as well?

File: src/data/bucket/circle_bucket.ts [hunk 1]
```diff
@@ -116,7 +115,7 @@ class CircleBucket<Layer extends CircleStyleLayer | HeatmapStyleLayer> implement
         if (sortFeaturesByKey) {
             bucketFeatures.sort((a, b) => {
                 // a.sortKey is always a number when in use
-                return (a.sortKey as any as number) - (b.sortKey as any as number);
+                return (a.sortKey) - (b.sortKey);
```
Comment (chippieTV) [inline src/data/bucket/circle_bucket.ts:118]: I realize I can remove the parens.. will cleanup some other stuff and add to this PR

Comment (chippieTV) [reply src/data/bucket/circle_bucket.ts:118]: are you ok with
```typescript
if (sortFeaturesByKey) {
    bucketFeatures.sort((a, b) => a.sortKey - b.sortKey);
}
```

or you prefer to keep it
```typescript
if (sortFeaturesByKey) {
    bucketFeatures.sort((a, b) => {
        return a.sortKey - b.sortKey;
    });
}
```

not sure the line between types only and cleaning up syntax.. maybe it's less consistent

**Comment by chippieTV** [comment]
also adding some fixes for broken formatting.

I don't want this to get too opinionated, so checking preferences or if anyone knows of some rule specified somewhere?

There are issues of 'bad formatting' which was probably automatic in TS conversion, but we're left with a couple of options. It's nothing too important I just don't want to seem like I'm making the rules here.

```typescript
// example original
glyphMap: {
  [_: string]: {
    [_: number]: StyleGlyph | undefined | null
  }
},

// example oneline
glyphMap: {[_: string]: {[_: number]: StyleGlyph | undefined | null}},
```

Me personally I think it's a bit too dense and there are cases of deeper nested objects where it gets a bit silly. I'm also more used to seeing spaces after braces but the whole project doesn't do that so I'm not going to randomly introduce it in the few places I touch now.

Depending on the specific code these different types will not be there so this example may have more noise than usual.

Thoughts?

cc/ @HarelM @JannikGM (your comment on bad formatting)

**Comment by chippieTV** [comment]
changing to draft.. will check that these changes relate to typescript branch changes only and don't update formatting in main branch causing more changes to review in typescript branch.

**Comment by HarelM** [comment]
Can I merge this? Is this done?

**Pull request merged**